planner/core/rule_aggregation_push_down.go

```diff
@@ -500,3 +513,70 @@ func (a *aggregationPushDownSolver) aggPushDown(p LogicalPlan, opt *logicalOptim
 func (*aggregationPushDownSolver) name() string {
 	return "aggregation_push_down"
 }
+
+func appendAggPushDownAcrossJoinTraceStep(oldAgg, newAgg *LogicalAggregation, aggFuncs []*aggregation.AggFuncDesc, join *LogicalJoin,
+	childIdx int, opt *logicalOptimizeOp) {
+	reason := func() string {
+		buffer := bytes.NewBufferString("agg[%v]'s functions[")
+		for i, aggFunc := range aggFuncs {
+			if i > 0 {
+				buffer.WriteString(",")
+			}
+			buffer.WriteString(aggFunc.String())
+		}
+		buffer.WriteString("] are decomposable with join")
+		return buffer.String()
+	}()
```

Comment:
Why do we use functions here and below?

Reply:
To build reason/action in a separate block.

Reply:
Ok...though I think we don't need a function to achieve this.